kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/installation/kroxylicious/Kroxylicious.java

```diff
@@ -92,17 +96,27 @@ public void deployPortIdentifiesNodeWithNoFilters(String clusterName) {
      * @param clusterName the cluster name
      */
     public void deployPortIdentifiesNodeWithTlsAndNoFilters(String clusterName) {
-        createCertificateConfigMap(deploymentNamespace);
+        Tls tls = createCertificateConfigMapFromListener(deploymentNamespace);
         resourceManager.createResourceFromBuilder(
                 KroxyliciousKafkaProxyTemplates.defaultKafkaProxyCR(deploymentNamespace, Constants.KROXYLICIOUS_PROXY_SIMPLE_NAME),
                 KroxyliciousKafkaProxyIngressTemplates.defaultKafkaProxyIngressCR(deploymentNamespace, Constants.KROXYLICIOUS_INGRESS_CLUSTER_IP,
                         Constants.KROXYLICIOUS_PROXY_SIMPLE_NAME),
-                KroxyliciousKafkaClusterRefTemplates.kafkaClusterRefCRWithTls(deploymentNamespace, clusterName),
+                KroxyliciousKafkaClusterRefTemplates.kafkaClusterRefCRWithTls(deploymentNamespace, clusterName, tls),
                 KroxyliciousVirtualKafkaClusterTemplates.defaultVirtualKafkaClusterCR(deploymentNamespace, clusterName, Constants.KROXYLICIOUS_PROXY_SIMPLE_NAME,
                         clusterName, Constants.KROXYLICIOUS_INGRESS_CLUSTER_IP));
     }
 
-    private void createCertificateConfigMap(String namespace) {
+    public void deployPortIdentifiesNodeWithDownstreamTlsAndNoFilters(String clusterName, Tls tls) {
+        resourceManager.createResourceFromBuilder(
+                KroxyliciousKafkaProxyTemplates.defaultKafkaProxyCR(deploymentNamespace, Constants.KROXYLICIOUS_PROXY_SIMPLE_NAME),
+                KroxyliciousKafkaProxyIngressTemplates.tlsKafkaProxyIngressCR(deploymentNamespace, Constants.KROXYLICIOUS_INGRESS_CLUSTER_IP,
+                        Constants.KROXYLICIOUS_PROXY_SIMPLE_NAME),
+                KroxyliciousKafkaClusterRefTemplates.kafkaClusterRefCRWithTls(deploymentNamespace, clusterName, tls),
```

Comment:
Just from eyeballing, haven't tried, it looks like upstream and downstream TLS are configured. Changing the secret will be updating the annotation on the KafkaService and triggering redeploy. To see the problem we need downstream TLS with a plain upstream connection.

Reply:
Ahhhh! Ok.

I'll tweak it later.

Reply:
yeah that was it.